src/common/com/intellij/plugins/haxe/util/HaxeResolveUtil.java

```diff
@@ -424,7 +424,7 @@ public static HaxeClassResolveResult getHaxeClassResolveResult(@Nullable PsiElem
     if (element instanceof HaxeComponentName) {
       return getHaxeClassResolveResult(element.getParent(), specialization);
     }
-    if (element instanceof AbstractHaxeTypeDefImpl) {
+    if (element instanceof HaxeTypedefDeclaration) {
```

Comment:
Since it's a guard for a specific cast, I think you should revert it.  What I mean is: If you are going to cast a type and you use instanceof to test for the correct type, then the test and the cast should be the same.  Otherwise, the reason for the instanceof test is confusing.  

Secondly, if you have to cast to an implementation type, then you must be calling a method that isn't defined by the interface -- therefore, checking whether the class is of an interface type does not tell you if the method you want to call is present.